exporter/kafkaexporter/converter.go

```diff
@@ -0,0 +1,106 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafkaexporter
+
+import (
+	"github.com/Shopify/sarama"
+)
+
+// MessageConverter converts our message type to sarama message type
+type MessageConverter interface {
```

Comment:
I think we can make this better by changing the "marshaler" interface to produce `[]*sarama.ProducerMessage` directly? What do you think? In that case we skip the `Message` creation (unnecessary allocation for the slice of messages).

Reply:
> I think we can make this better by changing the "marshaler" interface to produce `[]*sarama.ProducerMessage` directly? What do you think? In that case we skip the `Message` creation (unnecessary allocation for the slice of messages).

During my implementation it did feels kind of tedious to create another "converter" interface to adjust such a simple behavior. I agree that changing the marshaler interface to generate `samara.ProducerMessage` directly seems like a better idea. I will make the modification accordingly.